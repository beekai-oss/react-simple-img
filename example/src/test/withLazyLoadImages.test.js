import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import withLazyLoadImages from '../lazyLoadImages/withLazyLoadImages';

const TestComponent = () => <div>Test</div>;
const IntersectionObserverSpy = jest.fn();
let IntersectionObserver;

describe('withLazyLoadImages', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    IntersectionObserver = global.IntersectionObserver;
    global.IntersectionObserver = IntersectionObserverSpy;
  });

  afterEach(() => {
    global.IntersectionObserver = IntersectionObserver;
  });

  it('should render correctly', () => {
    const Component = withLazyLoadImages(TestComponent);
    const tree = renderer.create(<Component />);
    expect(tree).toMatchSnapshot();
  });

  it('should initialise IntersectionObserver and observe each images', () => {
    const Component = withLazyLoadImages(TestComponent);

    mount(<Component />);

    expect(IntersectionObserverSpy).toHaveBeenCalled();
  });

  it('should append image into images', () => {
    const Component = withLazyLoadImages(TestComponent);
    const tree = mount(<Component />);
    tree.instance().appendImageRef('image');
    tree.update();

    expect(tree.instance().images).toEqual(['image']);
  });

  it('should remove image from images', () => {
    const Component = withLazyLoadImages(TestComponent);
    const tree = mount(<Component />);
    tree.instance().removeImageRef('image');
  });
});