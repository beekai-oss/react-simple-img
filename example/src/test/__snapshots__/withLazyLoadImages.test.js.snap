// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withLazyLoadImages should render correctly 1`] = `
<div>
  Test
</div>
`;

exports[`withLazyLoadImages when preloadImage called should catch error failed 1`] = `[Error: ðŸ’© Fetch image failed with target {} and error message TypeError: Cannot read property 'src' of undefined]`;
